### DISCUSSION TITLE
idea: Successful individual dep updates batched into a grouped update

### DISCUSSION THREAD

**Discussion by msavy** [renovatebot/renovate#10355] [Suggest an Idea]
To avoid a huge amount of update noise filling up the git history for minor & patch dependency updates, I have a workflow that I use with Dependabot where I: 

* Let Dependabot open individual PRs for all dependency updates.
* Those that pass successfully I batch together into a single PR (currently a manual process via cherry-pick).
  * [This gets squashed into a single PR, in a very similar way to Renovate's grouped updates](https://github.com/apiman/apiman/commit/e605ecdeb07f6f7df8f36b8b5b7b2a85748d4c94).
* Those that fail, go down an exceptional handling path (uninteresting for the purposes of this conversation).

I find this better than just batching from the beginning, as you can identify failures and exclude/pin/ignore certain packages rather easily.

It strikes me that this could be an improvement that utilises much of the existing underlying functionality in renovate.

Perhaps it could even be bot/comment-driven. For example, on a PR I want to batch, I could say `@renovate add this to batch #1234`. It could certainly be config-driven as well.

**Comment by rarkins** [comment]
I'm working on something similar currently: https://github.com/renovatebot/renovate/issues/4404

The goal would be to make it essentially hands-free which would suit the majority of users best.

**Reply by msavy** [reply]
If you think this is redundant, please feel free to close it out.

I do wonder whether some commands would be quite useful, particularly for handling edge cases, for example if I want to eject a particular dependency from a roll-up patch for whatever reason.

**Reply by msavy** [reply]
And thank you for your wonderful work. I do know that open source development can be a rather thankless task much of the time :-).

**Reply by rarkins** [reply]
This is the correct place for this topic, so no need to do anything. My plan was to see how the planned feature works for people in practice then iterate on it

**Reply by msavy** [reply]
Fantastic. Happy to give it a try when it's ready and provide feedback.
